compute_sdk/globus_compute_sdk/sdk/web_client.py

```diff
@@ -196,6 +197,8 @@ def register_endpoint(
             data["allowed_functions"] = allowed_functions
         if auth_policy:
             data["authentication_policy"] = auth_policy
+        if subscription_id:
+            data["subscription_uuid"] = subscription_id
```

Comment:
Should this path be tested by the SDK?  I see that the _endpoint_ effectively tests this path, but nominally the responsibility to do rests with the SDK, no?